core/client/src/main/java/alluxio/client/file/FileOutStream.java

```diff
@@ -116,7 +120,8 @@ public FileOutStream(AlluxioURI path, OutStreamOptions options) throws IOExcepti
         String tmpPath = PathUtils.temporaryFileName(mNonce, mUfsPath);
         UnderFileSystem ufs = UnderFileSystem.get(tmpPath, ClientContext.getConf());
         // TODO(jiri): Implement collection of temporary files left behind by dead clients.
-        mUnderStorageOutputStream = ufs.create(tmpPath, (int) mBlockSize);
+        CreateOptions createOptions = new CreateOptions().setPermission(options.getPermission());
```

Comment:
I think it makes more sense for non-wire types to use `Permission` (similar to how `OutStreamOptions` does it) and do the packing / unpacking only when sending / receving things over the wire ... I don't feel strongly about this though so I will leave up to you